### PR TITLE
feat: Move advance req get to platform API

### PR DESCRIPTION
**Pull request by bistaastha**
Fixing tests here: https://github.com/fylein/fyle-mobile-app/pull/2855

### CODE REVIEW

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 1]
```diff
@@ -24,14 +24,15 @@ import { OrgSettingsService } from 'src/app/core/services/org-settings.service';
...
+import { ExtendedAdvanceRequestPublic } from 'src/app/core/models/extended-advance-request-public.model';
 
 @Component({
   selector: 'app-my-advances',
   templateUrl: './my-advances.page.html',
   styleUrls: ['./my-advances.page.scss'],
 })
 export class MyAdvancesPage implements AfterViewChecked {
-  myAdvanceRequests$: Observable<ExtendedAdvanceRequest[]>;
+  myAdvanceRequests$: Observable<ExtendedAdvanceRequestPublic[]>;
```
Comment (bistaastha) [inline src/app/fyle/my-advances/my-advances.page.ts:35]: Previous model cannot be updated as team advance is using it

File: src/app/fyle/my-view-advance-request/my-view-advance-request.page.ts [hunk 2]
```diff
@@ -21,14 +21,15 @@ import { FyPopoverComponent } from 'src/app/shared/components/fy-popover/fy-popo
 import { StatisticTypes } from 'src/app/shared/components/fy-statistic/statistic-type.enum';
 import { getCurrencySymbol } from '@angular/common';
 import { ExpenseFieldsService } from 'src/app/core/services/expense-fields.service';
+import { ExtendedAdvanceRequestPublic } from 'src/app/core/models/extended-advance-request-public.model';
```
Comment (mvaishnavi) [inline src/app/fyle/my-view-advance-request/my-view-advance-request.page.ts:24]: we should remove the import for `ExtendedAdvanceRequest`

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 3]
```diff
@@ -24,14 +24,15 @@ import { OrgSettingsService } from 'src/app/core/services/org-settings.service';
 import { ExtendedAdvanceRequest } from 'src/app/core/models/extended_advance_request.model';
 import { ExtendedAdvance } from 'src/app/core/models/extended_advance.model';
 import { MyAdvancesFilters } from 'src/app/core/models/my-advances-filters.model';
+import { ExtendedAdvanceRequestPublic } from 'src/app/core/models/extended-advance-request-public.model';
```
Comment (mvaishnavi) [inline src/app/fyle/my-advances/my-advances.page.ts:27]: remove the other import from here.

**Review by mvaishnavi — APPROVED** [review]

File: src/app/core/services/utility.service.ts [hunk 4]
```diff
@@ -153,9 +154,9 @@ export class UtilityService {
     return result;
   }
 
-  private getSortingValue(advance: ExtendedAdvanceRequest, sortParam: SortingParam): dayjs.Dayjs | string {
+  private getSortingValue(advance: ExtendedAdvanceRequestPublic, sortParam: SortingParam): dayjs.Dayjs | string {
     if (sortParam === SortingParam.creationDate) {
-      return advance.areq_created_at ? dayjs(advance.areq_created_at) : dayjs(advance.adv_created_at);
```
Comment (bistaastha) [inline src/app/core/services/utility.service.ts:158]: areq_created_at won't be null for advance requests

**Review by suyashpatil78 — APPROVED** [review]
Please to take care of one comment! Rest LGTM

File: src/app/core/services/utility.service.ts (hunk 4 above)
Comment (Chethan-Fyle) [reply src/app/core/services/utility.service.ts:158]: Are you sure? Because admin can issue advance without users creating an advance request.

**Review by Chethan-Fyle — APPROVED** [review]
One minor comment.

File: src/app/core/services/utility.service.ts (hunk 4 above)
Comment (bistaastha) [reply src/app/core/services/utility.service.ts:158]: Advances are handled separately, the type here is only for Advance requests

**Pull request merged**